webui/react/src/hooks/useModal/useModalChangePassword.tsx

```diff
@@ -0,0 +1,122 @@
+        password: form.getFieldValue('newPassword'),
+        username,
+      });
+      modalClose();
+    };
+
+    return (
+      <div className={css.base}>
+        <Form form={form} layout="vertical" onFinish={submitForm}>
+          <Form.Item
+            label="Old Password"
+            name="oldPassword"
+            required
+            rules={[
+              {
+                message: 'Incorrect password',
+                validator: async (rule, value) => {
+                  return await login({
+                    password: value || '',
+                    username: username || '',
+                  });
+                },
+              },
+            ]}
+            validateTrigger="onSubmit">
+            <Input.Password />
+          </Form.Item>
+          <Form.Item
+            label="New Password"
+            name="newPassword"
+            rules={[
+              { message: 'New password required', required: true },
+              { message: "Your new password isn't long enough", min: 8 },
+              {
+                message: 'Your new password must include an uppercase',
+                pattern: /[A-Z]+/,
+              },
+              {
+                message: 'Your new password must include a lowercase',
+                pattern: /[a-z]+/,
+              },
+              {
+                message: 'Your new password must include a number',
+                pattern: /\d/,
+              },
+            ]}>
+            <Input.Password />
+          </Form.Item>
+          <Form.Item
+            dependencies={[ 'newPassword' ]}
+            label="Confirm Password"
+            name="confirmPassword"
+            rules={[
+              { message: 'Confirmed password required', required: true },
+              {
+                message: 'Your new passwords do not match',
+                validator: (rule, value) => {
+                  return value === form.getFieldValue('newPassword')
+                    ? Promise.resolve()
+                    : Promise.reject();
+                },
+              },
+            ]}>
+            <Input.Password />
+          </Form.Item>
+          <Form.Item>
+            <span>
+              Password must be at least 8 charactes and contain one uppercase,
+              lowercase.
+            </span>
+          </Form.Item>
+          <Form.Item>
+            <div className={css.buttons}>
+              <Button onClick={() => onCancel()}>Cancel</Button>
+              <Button htmlType="submit" type="primary">
+                Change Password
+              </Button>
+            </div>
+          </Form.Item>
+        </Form>
+      </div>
+    );
+  };
+
+  const onCancel = () => {
+    form.resetFields();
+    modalClose();
+  };
+
+  const modalOpen = () => {
+    openOrUpdate({
+      className: css.noFooter,
+      closable: true,
+      content: getModalContent(),
+      icon: null,
+      onCancel,
+      title: 'Change Password',
+    });
+  };
```

Comment:
nit: these should both be useCallback functions unless there's a reason they can't be
question: I see that you're removing the footer here in CSS, but why, when you're just re-implementing it yourself?